packages/adapters/ethers/src/client.ts

```diff
@@ -175,889 +95,394 @@ export class EthersAdapter {
+    const result = await EthersMethods.writeContract(
+      {
+        abi: params.abi,
+        method: params.method,
+        fromAddress: params.caipAddress as `0x${string}`,
+        receiverAddress: params.receiverAddress as `0x${string}`,
+        tokenAmount: params.tokenAmount,
+        tokenAddress: params.tokenAddress as `0x${string}`
       },
+      params.provider as Provider,
+      params.caipAddress,
+      Number(params.caipNetwork?.id)
+    )
 
-      sendTransaction: async data => {
-        if (data.chainNamespace && data.chainNamespace !== 'eip155') {
-          throw new Error(`Invalid chain namespace - Expected eip155, got ${data.chainNamespace}`)
-        }
-        const provider = ProviderUtil.getProvider<Provider>('eip155')
-        const caipAddress = ChainController.state.activeCaipAddress
-        const address = CoreHelperUtil.getPlainAddress(caipAddress)
-        const caipNetwork = this.appKit?.getCaipNetwork()
-
-        if (!address) {
-          throw new Error('Address is undefined')
-        }
+    return { hash: result }
+  }
 
-        if (!provider) {
-          throw new Error('Provider is undefined')
-        }
+  public async estimateGas(
+    params: AdapterBlueprint.EstimateGasTransactionArgs
+  ): Promise<AdapterBlueprint.EstimateGasTransactionResult> {
+    const { provider, caipNetwork, address } = params
+    if (!provider) {
+      throw new Error('Provider is undefined')
+    }
 
-        return await EthersMethods.sendTransaction(data, provider, address, Number(caipNetwork?.id))
-      },
+    try {
+      const result = await EthersMethods.estimateGas(
+        {
+          data: params.data as `0x${string}`,
+          to: params.to as `0x${string}`,
+          address: address as `0x${string}`
+        },
+        provider as Provider,
+        address as `0x${string}`,
+        Number(caipNetwork?.id)
+      )
 
-      writeContract: async data => {
-        const provider = ProviderUtil.getProvider<Provider>('eip155')
-        const caipAddress = ChainController.state.activeCaipAddress
-        const address = CoreHelperUtil.getPlainAddress(caipAddress)
-        const caipNetwork = this.appKit?.getCaipNetwork()
+      return { gas: result }
+    } catch (error) {
+      throw new Error('EthersAdapter:estimateGas - Estimate gas failed')
+    }
+  }
 
-        if (!address) {
-          throw new Error('Address is undefined')
-        }
+  public async getEnsAddress(
+    params: AdapterBlueprint.GetEnsAddressParams
+  ): Promise<AdapterBlueprint.GetEnsAddressResult> {
+    const { name, appKit } = params
+    if (appKit) {
+      const result = await EthersMethods.getEnsAddress(name, appKit)
 
-        if (!provider) {
-          throw new Error('Provider is undefined')
-        }
+      return { address: result }
+    }
 
-        return await EthersMethods.writeContract(data, provider, address, Number(caipNetwork?.id))
-      },
+    return { address: false }
```

Comment:
is this ok? I know it was like this before but wouldn't having an empty string here be a more logical response?